### ISSUE TITLE
inner_product_scan plots extra axis

### ISSUE THREAD

**Issue by jrmlhermitte** [NSLS-II/bluesky#961]
```py
RE(inner_product_scan([hutchb_diag], 11, jj_slits.inboard, -2, 2, jj_slits.outboard, -1.8, 
    ...: -1.8+4), hints={'fields': [hutchb_diag.stats1.total.name], 'dimensions' : [([jj_slits.outbo
    ...: ard.name, jj_slits.inboard.name], 'primary')]})

```
plots an extra axis:

<img src="https://user-images.githubusercontent.com/705366/35359611-c8bedeb8-0128-11e8-970a-c74194c8f463.png" width=200 alt="hutchb_diag_stats1_total_jj_slits_inboard_vs_jj_slits_outboard-1" />

Only the bottom plot should be present.

**Comment by jrmlhermitte** [comment]
it seems the issue was resolved for relative scans but not absolute. i think the BEC hints logic should move from relative to absolute

**Comment by jrmlhermitte** [comment]
Here is a script to reproduce results, and the plots one by one:
```python
from bluesky.plans import (relative_outer_product_scan,
                           relative_inner_product_scan,
                           outer_product_scan,
                           inner_product_scan)
from bluesky.plan_stubs import mov

from ophyd.sim import det4, motor1, motor2
from bluesky import RunEngine
from bluesky.callbacks.best_effort import BestEffortCallback

RE = RunEngine({})
# subscribe BEC
bec = BestEffortCallback()
RE.subscribe(bec)


# move motor to a reproducible location
RE(mov(motor1, 0))
RE(mov(motor2, 0))
RE(relative_outer_product_scan([det4], motor1, -1, 1, 20, motor2, -2, 2, 20, True))
RE(outer_product_scan([det4], motor1, -1, 1, 20, motor2, -2, 2, 20, True))

# move motor to a reproducible location
RE(mov(motor1, 0))
RE(mov(motor2, 0))
RE(relative_inner_product_scan([det4], 20, motor1, -1, 1, motor2, -2, 2))
RE(inner_product_scan([det4], 20, motor1, -1, 1, motor2, -2, 2))
```

![bec_outer_1](https://user-images.githubusercontent.com/705366/35361874-fe251ae2-0130-11e8-95cf-758c7906c04b.png)
![bec_outer_2](https://user-images.githubusercontent.com/705366/35361875-01ab4268-0131-11e8-95af-ac0a93e762ab.png)
![bec_relative](https://user-images.githubusercontent.com/705366/35361886-080165b6-0131-11e8-9304-0e7552ba034f.png)
![bec_absolute](https://user-images.githubusercontent.com/705366/35361890-0d0384c2-0131-11e8-8f55-f107a7d9cb97.png)

**Comment by danielballan** [comment]
Closed by #963
